SMLHelper/Patchers/ItemsContainerPatcher.cs

```diff
@@ -1,174 +1,76 @@
-            // How does the base game get away with calculating this mess on every frame???
-            HasRoomCacheCollection[container][itemSize] = hasRoom;
-
-            if (hasRoom && // True means there is space for an item of this size
-                (itemSize.x > 1 || itemSize.y > 1)) // If there is room for an item of a size larger than a 1x1
-            {
-                // Then we can preemptively add all item sizes that are smaller.
-                foreach (Vector2int smallerItem in AllSmallerThan(itemSize))
-                    HasRoomCacheCollection[container][smallerItem] = true;
-            }
-        }
-
-        private static void NotifyChangeItem_Postfix(ItemsContainer __instance, InventoryItem item)
-        {
-            // Items in the inventory have changed. Clear out the cache.
-            if (HasRoomCacheCollection.TryGetValue(__instance as IItemsContainer, out Dictionary<Vector2int, bool> cache))
-            {
-                cache.Clear();
-            }
-            else
-            {
-                // This is a new container we haven't seen before, save it to the cache collection.
-                HasRoomCacheCollection.Add(__instance as IItemsContainer, new Dictionary<Vector2int, bool>());
-            }
-        }
-
-        private static readonly IEnumerable<Vector2int> SmallerThan3x3 = new Vector2int[]
-        {
-            new Vector2int(2,3),
-            new Vector2int(3,2),
-            new Vector2int(2,2),
-            new Vector2int(1,2),
-            new Vector2int(2,1),
-            new Vector2int(1,1)
-        };
-
-        private static readonly IEnumerable<Vector2int> SmallerThan2x3 = new Vector2int[]
-        {
-            new Vector2int(2,2),
-            new Vector2int(1,2),
-            new Vector2int(2,1),
-            new Vector2int(1,1)
-        };
-
-        private static readonly IEnumerable<Vector2int> SmallerThan2x2 = new Vector2int[]
-        {
-            new Vector2int(1,2),
-            new Vector2int(2,1),
-            new Vector2int(1,1)
-        };
-
-        private static readonly IEnumerable<Vector2int> Just1x1 = new Vector2int[]
-        {
-            new Vector2int(1,1)
-        };
-
-        private static IEnumerable<Vector2int> AllSmallerThan(Vector2int original)
-        {
-            // This might not be pretty but it's simple enough to keep fast
-
-            if (original.x == 3 && original.y == 3) // Size of Mobile Vehicle Bay
-            {
-                return SmallerThan3x3;
-            }
-
-            if (original.x == 2 && original.y == 3) // Size of the Seaglide
-            {
-                return SmallerThan2x3;
-            }
-
-            if (original.x == 2 && original.y == 2) // Size of many things bigger than 1x1
-            {
-                return SmallerThan2x2;
-            }
-
-            // 1x2 is the size of a Creature Decoy
-
-            return Just1x1;
-        }
-    }
+	using System;
```

Comment:
The indentation triggers my OCD. Please use spaces instead of tabs

Reply:
I'll clean this up later on another branch.